### PR TITLE
WT-5655 Avoid creating a tombstone when we already have a tombstone

### PR DESCRIPTION
**Pull request by luke-pearson**

### CODE REVIEW

File: src/reconcile/rec_visibility.c [hunk 1]
```diff
@@ -114,13 +114,14 @@ __rec_append_orig_value(
...
          * We need to append a TOMBSTONE before the onpage value if the onpage value has a valid
-         * stop pair.
+         * stop pair. Except when the onpage value is a tombstone.
          *
          * Imagine a case we insert and delete a value respectively at timestamp 0 and 10, and later
          * insert it again at 20. We need the TOMBSTONE to tell us there is no value between 10 and
          * 20.
          */
-        if (unpack->stop_ts != WT_TS_MAX || unpack->stop_txn != WT_TXN_MAX) {
+        if (upd->type != WT_UPDATE_TOMBSTONE &&
```
Comment (quchenhao) [inline src/reconcile/rec_visibility.c]: Can we at least assert the stop time pair match the tombstone?

Comment (luke-pearson) [reply src/reconcile/rec_visibility.c]: Yep I'll add that in.

**Comment by quchenhao** [comment]
We can also enable the join_visibility test.

File: test/evergreen.yml [hunk 2]
```diff
@@ -1980,7 +1979,7 @@ tasks:
           test_env_vars: ASAN_OPTIONS="detect_leaks=1:abort_on_error=1:disable_coredump=0" ASAN_SYMBOLIZER_PATH=/opt/mongodbtoolchain/v3/bin/llvm-symbolizer
           # run for 24 hours ( 24 * 60 = 1440 minutes), skip known errors, don't stop at failed tests, use default config
           format_test_script_args: -E -t 1440
-
```
Comment (luke-pearson) [inline test/evergreen.yml]: Just removed some extra whitespace.

File: test/evergreen.yml [hunk 3]
```diff
@@ -2184,7 +2183,7 @@ tasks:
         vars:
           # CC is set to the system default "clang" binary here as a workaround. Change it back to mongodbtoolchain "clang" binary after BUILD-10248 fix.
           configure_env_vars: CCAS=/opt/mongodbtoolchain/v3/bin/gcc CC=/usr/bin/clang CXX=/opt/mongodbtoolchain/v3/bin/clang++ PATH=/opt/mongodbtoolchain/v3/bin:$PATH CFLAGS="-ggdb -fPIC -fsanitize=address -fno-omit-frame-pointer -I/opt/mongodbtoolchain/v3/lib/gcc/ppc64le-mongodb-linux/8.2.0/include"
-          posix_configure_flags: --enable-diagnostic --with-builtins=lz4,snappy,zlib
```
Comment (luke-pearson) [inline test/evergreen.yml]: And here.

File: src/reconcile/rec_visibility.c [hunk 4]
```diff
@@ -132,6 +133,13 @@ __rec_append_orig_value(
             tombstone->next = append;
             total_size += size;
         }
+        /*
+         * Assert that if we did have a tombstone as the original update that its start timestamp
+         * matches the previous updates stop timestamp. Note that the stop_ts on the previous update
+         * can be WT_TS_MAX so we should ignore this.
+         */
+        WT_ASSERT(session, upd->type != WT_UPDATE_TOMBSTONE || unpack->stop_ts == WT_TS_MAX ||
+            unpack->stop_ts == upd->start_ts);
```
Comment (quchenhao) [inline src/reconcile/rec_visibility.c]: Why not assert transaction id matches? In this case, we shouldn't clear the transaction ids so it should match.

Comment (luke-pearson) [reply src/reconcile/rec_visibility.c]: The transaction IDs don't match if you look at the core I put in WT-5655, unless I'm missing something.

Comment (quchenhao) [reply src/reconcile/rec_visibility.c]: May be something like this?
#ifdef HAVE_DIAGNOSTICS
else if (upd ->type == WT_UPDATE_TOMBSTONE)
      WT_ASSERT(session, unpack->stop_txn == upd->txnid && unpack->start_ts == upd->start_ts);
#5247

**Pull request closed without merging**